README
Put dvdauthor.py, Pydvdauthor.py, Pydvdauthor_curses.py, and Pydvdauthor_wxPython.py in the same directory.

Then what you type depends on what type of interface you want.  If you want A completely text-based interface, type:

python Pydvdauthor.py

If you want a semi-GUI in the commandline via ncurses, type:

python Pydvdauthor_curses.py

If you want a full-fledged GUI, type:

python Pydvdauthor_wxPython.py

ENJOY!

Changelog
Changelog

5 Feb 2008
v0.1.6
-added ncurses interface and GUI via wxPython

10 May 2006
v0.1.5
-via while loop, the program will now keep asking the user if everything is correct until he or she says it is

5 May 2006
v0.1
Created the first version and set things up


dvdauthor.py
import os, sys

def dvdauthorxml(pathtodvd, pathtovideo, chapters):
    """This will take in 3 parameters:
    The path where the DVD will be created, the path to the video which will make up the DVD, and the times where chapters should be created.
    This is meant to be the backend which creates the dvdauthor.xml file.
    It is up to the programmer to design a frontend for user interaction.

    pathtodvd and pathtovideo should be strings and chapters should be a list

    This currently creates a DVD without a menu, it just begins playing the
    movie and the user can skip around the chapters."""

    __author__ = "Eric Mesa"
    __version__ = "v0.1.6"
    __license__ = "GNU GPL v3.0"
    __copyright__ = "(c) 2008 Eric Mesa"
    __email__ = "ericsbinaryworld at gmail dot com"

    outputfile = open('/home/emesa/bin/python/dvdauthor.xml', 'w')
    outputfile.write('<dvdauthor dest="' + pathtodvd + '"> \
\n<vmgm /> \
\n<titleset> \
\n<titles> \
\n<pgc> \
\n<vob file="' + pathtovideo  + '" chapters="' + ", ".join(chapters)  +  '"/> \
\n</pgc> \
\n</titles> \
\n</titleset> \
\n</dvdauthor>')
    outputfile.close()

    
#Testing Suite with the if name trick

if __name__ == "__main__":
    print "It worked!"
    x = 'pathtofile'
    y = 'pathtodvd'
    z = ['0', '14:37',]
    dvdauthorxml(y,x,z)


Pydvdauthor.py
import sys

"""Calls dvdauthorxml from dvdauthor.py to provide the information necessary to create the XML file."""

__program_name__ = "Pydvdauthor"
__author__ = "Eric Mesa"
__version__ = "v0.1.6"
__license__ = "GNU GPL v3"
__copyright__ = "(c)2008 Eric Mesa"
__email__ = "ericsbinaryworld at gmail dot com"

from dvdauthor import dvdauthorxml

print "Welcome to " + __author__ + "'s" + __program_name__
print "Version " + __version__
print __license__ + "\n"  + __copyright__

loop_me = True

#Begin the inquisition - no one EVER expects the inquisition!

print "Please enter the path where you want the DVD files created.  This must be a full path (eg /home/user_name/spamalot)"
pathtodvd = raw_input()

print "Please enter the path and filename where the video can be found.  At this time, this must be an MPEG-2 DVD compliant video (eg /home/user/spamspamspam.mpg)"
pathtovideo = raw_input()

print "Please enter the times where you want a chapter to be created.  \n Enter this in the format of \n hour:minute:seconds.  \n If you don't want any chapters then at least enter 0. \n Separate all chapters with a comma. (eg 0, 14:30, 19:34)"
chapters = raw_input()

#Hit them with a fish, I mean, with what they have given us so far

print "Here is what you have so far:"
print pathtodvd + "\n" + pathtovideo + "\n" + chapters + "\n"

#Ask if it's correct

print "\n\nIs this correct? (y or n)"
q1 = raw_input()

#A, currently, very untidy little bugger which only allows them to correct one mistake

if q1 == 'y':
    print "Excellent!  We will continue!"
    loop_me = False

while loop_me == True:
    print "Which was wrong?  Your choices are (dvd, file, chapters)"
    q2 = raw_input()

    if q2 == 'dvd':
        print "Enter the path where you want the dvd created"
        pathtodvd = raw_input()
    elif q2 == 'file':
        print "Enter the path where the file is:"
        pathtovideo = raw_input()
    elif q2 == 'chapters':
        print "Re-enter the chapters."
        chapters = raw_input()
    else:
        print "Please give a valid reponse!"

    print "Here is what you have now entered:"
    print "Here is what you have so far:"
    print pathtodvd + "\n" + pathtovideo + "\n" + chapters + "\n"
    
    print "Is everything correct now? (y or n)"
    q3 = raw_input()
    if q3 == 'y':
        loop_me = False
    

print "\nGenerating the XML file!"

chapters2 = chapters.split(",")

dvdauthorxml(pathtodvd, pathtovideo, chapters2)

print "Now look for the file named dvdauthor.xml.  It should be in this path, then type dvdauthor -x dvdauthor.xml"

#Yippie!


Pydvdauthor_curses.py
import os, sys, curses

"""Calls dvdauthorxml from dvdauthor.py to provide the information necessary to create the XML file."""

from dvdauthor import dvdauthorxml

__program_name__ = "Pydvdauthor_ncurses"
__author__ = "Eric Mesa"
__version__ = "v0.1.6"
__license__ = "GNU GPL v3"
__copyright__ = "(c)2008 Eric Mesa"
__email__ = "ericsbinaryworld at gmail dot com"

def get_param(prompt_string, pos):
    """This asks the user for a string and then returns to the program"""
    #screen.clear()
    #screen.border(0)
    screen.addstr(pos,2,prompt_string)
    screen.refresh()
    input = screen.getstr(pos+5,10,60)
    return input

def execute_cmd(cmd_string):
    """Executes a command"""
    system("clear")
    a = system(cmd_string)
    print ""
    if a == 0:
        print "Command executed correctly"
    else:
        print "Command terminated with error"
    raw_input("Press Enter")
    print ""

screen = curses.initscr()

screen.clear()
screen.border(0)
screen.addstr(2,2,"Welcome to " + __author__ + "'s " + __program_name__)
screen.addstr(3,2,"Version " + __version__)
screen.addstr(4,2,__license__)
screen.addstr(5,2, __copyright__)
screen.refresh()

#may want to change this - getch is get character
#screen.getch()

pathtodvd = get_param("Please enter the path where you want the DVD files created.  This must be a full path (eg /home/user_name/spamalot)", 6)

pathtovideo = get_param("Please enter the path and filename where the video can be found.  At this time, this must be an MPEG-2 DVD compliant video (eg /home/user/spamspamspam.mpg)", 14)

chapters = get_param( "Please enter the times where you want a chapter to be created.  \n Enter this in the format of \n hour:minute:seconds.  \n If you don't want any chapters then at least enter 0. \n Separate all chapters with a comma. (eg 0, 14:30, 19:34)", 22)

#do the part where they can change it - but that may not be necessary with the type of gui I want
#this will come later

screen.addstr(29,2, "Generating the XML file!")

screen.refresh()
screen.getch()
chapters2 = chapters.split(",")

dvdauthorxml(pathtodvd,pathtovideo,chapters2)

#screen.addstr(2,3, "Now look for the file named dvdauthor.xml.  It should be in this path, then type dvdauthor -x dvdauthor.xml")

curses.endwin()


Pydvdauthor_wxPython.py
import wx, os

from dvdauthor import dvdauthorxml

__program_name__ = "Pydvdauthor wxPython"
__author__ = "Eric Mesa"
__version__ = "v0.1.6"
__license__ = "GNU GPL v3"
__copyright__ = "(c)2008 Eric Mesa"
__email__ = "ericsbinaryworld at gmail dot com"

def cXML(event):
    ptdvd = pathtodvd.GetValue()
    ptvideo = pathtovideo.GetValue()
    chapters2 = chapters.GetValue().split(",")
    dvdauthorxml(ptdvd, ptvideo, chapters2)
    d = wx.MessageDialog( win, "Now look for the file named dvdauthor.xml.  It should be in this path, then type dvdauthor -x dvdauthor.xml","What to do next", wx.OK)
    d.ShowModal()
    d.Destroy()

ID_ABOUT = 101
ID_EXIT = 110

app = wx.App()
win = wx.Frame(None, title="Pydvdauthor", size=(500,500))

bkg = wx.Panel(win)

#the text boxes
pathtodvd = wx.TextCtrl(bkg)
pathtovideo = wx.TextCtrl(bkg)
chapters = wx.TextCtrl(bkg)

text1 = wx.StaticText(bkg, -1, "Please enter the path where you want the DVD files created.  This must be a full path (eg /home/user_name/spamalot)")
text2 = wx.StaticText(bkg, -1, "Please enter the path and filename where the video can be found.  At this time, this must be an MPEG-2 DVD compliant video (eg /home/user/spamspamspam.mpg)")
text3 = wx.StaticText(bkg, -1, "Please enter the times where you want a chapter to be created.  \n Enter this in the format of \n hour:minute:seconds.  \n If you don't want any chapters then at least enter 0. \n Separate all chapters with a comma. (eg 0, 14:30, 19:34)")

createXML = wx.Button(bkg, label="Create XML")
createXML.Bind(wx.EVT_BUTTON, cXML)

hbox1 = wx.BoxSizer()
hbox2 = wx.BoxSizer()
hbox3 = wx.BoxSizer()
hbox4 = wx.BoxSizer()
hbox5 = wx.BoxSizer()
hbox6 = wx.BoxSizer()
hbox7 = wx.BoxSizer()

hbox1.Add(text1, proportion=1, flag=wx.EXPAND)
hbox2.Add(pathtodvd, proportion=1, flag=wx.EXPAND)
hbox3.Add(text2, proportion=1, flag=wx.EXPAND)
hbox4.Add(pathtovideo, proportion=1, flag=wx.EXPAND)
hbox5.Add(text3, proportion=1, flag=wx.EXPAND)
hbox6.Add(chapters, proportion=1, flag=wx.EXPAND)
hbox7.Add(createXML, proportion=1, flag=wx.EXPAND)


vbox = wx.BoxSizer(wx.VERTICAL)
vbox.Add(hbox1, proportion=0, flag=wx.EXPAND | wx.ALL, border=5)
vbox.Add(hbox2, proportion=0, flag=wx.EXPAND | wx.ALL, border=5)
vbox.Add(hbox3, proportion=0, flag=wx.EXPAND | wx.ALL, border=5)
vbox.Add(hbox4, proportion=0, flag=wx.EXPAND | wx.ALL, border=5)
vbox.Add(hbox5, proportion=0, flag=wx.EXPAND | wx.ALL, border=5)
vbox.Add(hbox6, proportion=0, flag=wx.EXPAND | wx.ALL, border=5)
vbox.Add(hbox7, proportion=0, flag=wx.EXPAND | wx.ALL, border=5)

#menu and status bar
win.CreateStatusBar()
filemenu = wx.Menu()
filemenu.Append(ID_ABOUT, "&About", "Information About Pydvdauthor")
filemenu.AppendSeparator()
filemenu.Append(ID_EXIT, "&Exit", "Exit")
menuBar = wx.MenuBar()
menuBar.Append(filemenu, "&File")
win.SetMenuBar(menuBar)

def OnAbout(e):
    d = wx.MessageDialog( win, "A small utility to create dvdauthor xml file\n\n\n" + __program_name__ + "\nby\n" + __author__ + "\n" + __license__ + "\n"  + __copyright__, "About " + __program_name__ + __version__, wx.OK)
    d.ShowModal()
    d.Destroy()

def OnExit(e):
    win.Close(True)

#attach menu event to to method
wx.EVT_MENU(win, ID_ABOUT, OnAbout)
wx.EVT_MENU(win, ID_EXIT, OnExit)

bkg.SetSizer(vbox)

#show the window
win.Show()

#start the program
app.MainLoop()
